lms/djangoapps/courseware/masquerade.py

```diff
@@ -116,3 +174,146 @@ def get_masquerading_group_info(user, course_key):
     if not course_masquerade:
         return None, None
     return course_masquerade.group_id, course_masquerade.user_partition_id
+
+
+# Sentinel object to mark deleted objects in the session cache
+_deleted_sentinel = object()
+
+
+class MasqueradingFieldDataCache(object):
+    """
+    A wrapper for FieldDataCache used when masquerading as a different user.  Data is read from the
+    underlying cache, but written to the session, so the state of the actual user is not changed.
+    """
```

Comment:
@cpennington I refactored the code to use a `MasqueradingKeyValueStore` instead.  It required some refactoring in `module_render.py`, but those changes seem safe to me.

I also had to move `set_score()` off the `FieldDataCache`, since the cache is not available anymore at the place where the method is called, and it didn't really use the instance state anyway.  This change feels a bit dubious, since I moved a function from a place where it didn't really fit to a different place where it doesn't really fit.  I'd appreciate suggestions how to improve this.